apis/projectcontour/v1/httpproxy.go

```diff
@@ -171,6 +171,25 @@ type AuthorizationServer struct {
 	//
 	// +optional
 	FailOpen bool `json:"failOpen,omitempty"`
+
+	// BufferSettings specifies configuration for buffering request data sent to AuthorizationServer
+	// +optional
+	BufferSettings *AuthorizationServerBufferSettings `json:"bufferSettings,omitEmpty"`
+}
+
+// AuthorizationServerBufferSettings enables ExtAuthz filter to buffer client request data and send it as part of authorization request
+type AuthorizationServerBufferSettings struct {
+	// MaxRequestBytes sets the maximum size of message body ExtAuthz filter will hold in-memory.
+	// +optional
+	MaxRequestBytes uint32 `json:"maxRequestBytes,omitempty"`
```

Comment:
I will have to add validation to ensure this is greater than 0. 
Similar to validation in protobuf: https://github.com/envoyproxy/envoy/blob/dbb1f50ef340e565c5a495a17aac0afff5d9a4f9/api/envoy/extensions/filters/http/ext_authz/v3/ext_authz.proto#L158

Should I add it to internal/dag/httpproxy_processor.go ?

Reply:
we can use some `kubebuilder` validation here too to set defaults and minimum values

Reply:
Thanks @sunjayBhatia . Added kubebuilder validation and set minimum to be 1

Reply:
I think we want a *minimum* of 1, and a *default* of e.g. 1024:

```
// +kubebuilder:validation:Minimum=1
// +kubebuilder:default=1024

```

Reply:
Thanks! Fixed